Xamarin.PropertyEditing.Mac/Controls/EditorContainer.cs

```diff
@@ -24,6 +28,127 @@ public EditorViewModel ViewModel
 
 				if (PropertyButton != null)
 					PropertyButton.ViewModel = value as PropertyViewModel;
+
+				UpdateVariations (value as PropertyViewModel);
+			}
+		}
+
+		private void UpdateVariations (PropertyViewModel propertyViewModel)
+		{
+			if (propertyViewModel != null && propertyViewModel.Property.CanWrite) {
+				if (propertyViewModel.HasVariations) {
```

Comment:
This should be in the conditional above it. Otherwise if the property is writable but doesn't have variations, nothing will get cleaned up.

Reply:
True.